### ISSUE TITLE
Hack for iOS temporal inputs affecting OSX

### ISSUE THREAD

**Issue by mikaelharsjo** [twbs/bootstrap#16620]
`forms.less` includes hacks (line 180) for an iOS alignment bug. These get applied to Chrome and Safari on OSX as well on my retina screen MacBook. Giving for example `<input type="time">` an unexpected `line-height`
![screen shot 2015-06-11 at 10 48 14](https://cloud.githubusercontent.com/assets/1415572/8103327/e7b30ae6-1027-11e5-9478-7a5ef4835b41.png)

This is the rule I think needs tweaking:

```css
@media screen and (-webkit-min-device-pixel-ratio: 0) {
  input[type="date"],
  input[type="time"],
``` 
My `-webkit-min-device-pixel-ratio` is 2 so I'm well above this. I have no solution for this, sorry, and I don't think the original bug https://bugs.webkit.org/show_bug.cgi?id=139848 is fixed (I don't have an iOS device nearby).

**Comment by zacechola** [comment]
[JS Bin of issue](http://jsbin.com/kepare/1/edit?html,output), which I don't see in Chrome 45.0.2429.0 canary (64-bit) on OSX 10.10.3 or Safari version 8.0.6 (10600.6.3).

Could you provide a bit more browser/OS info and compare my bin to your where you see the issue?

X-Ref: #14234, #13098, #11266

**Comment by mitchlloyd** [comment]
This issue only seems to appear [when the `form-control` class is omitted](http://jsbin.com/kojosasedu/2/edit).

I just came across this when refactoring some markup. This works fine when using Bootstrap as intended, but it is a gotcha.

**Comment by cvrebert** [comment]
Testcase: http://jsfiddle.net/b8qbpxtq/2/
@mikaelharsjo I can't seem to reproduce (or otherwise I don't quite understand what the problem is).
Firefox and Chrome look like they style the text in the same way, and the text in the time inputs looks like it's styled the same as the text in the text inputs. This is on an OS X Yosemite retina MacBook Pro.
![shot](https://cloud.githubusercontent.com/assets/419884/8137221/9ab12bfe-10f6-11e5-943e-d61e63315b5f.png)

**Comment by mikaelharsjo** [comment]
@zacechola @mitchlloyd Maybe I was unclear but this is only a issue when you don't have `class="form-control"`. Even if that´s not the bootstrap way of doing it I would expect `<input type="time">` and `<input type="text">` next to each other be the same height.

**Comment by mikaelharsjo** [comment]
@cvrebert Your fiddle seems to work for me. Fixed in 3.3.5?

**Comment by mikaelharsjo** [comment]
Yes confirmed. This is resolved in master now by: 52110fa2c20d101e162179d7caf3a22992234d56
